src/components/display/fewlines/ValidateIdentityForm/ValidateIdentityForm.tsx

```diff
@@ -31,22 +60,26 @@ export const ValidateIdentityForm: React.FC<{
       <Form
         formID={formID}
         onSubmit={async () => {
-          await validateIdentity(validationCode).catch((error) => {
-            if (error instanceof InvalidValidationCode) {
-              setFormID(uuidv4());
-              setFlashMessage(error.message);
-            } else if (error instanceof TemporaryIdentityExpired) {
-              router && router.push("/account/logins/email/new");
-            } else {
-              throw error;
-            }
-          });
+          await validateIdentity(validationCode, eventId)
+            .then((path) => {
+              router && router.push(path);
+            })
+            .catch((error) => {
+              if (error instanceof InvalidValidationCode) {
+                setFormID(uuidv4());
+                setFlashMessage(error.message);
+              } else if (error instanceof TemporaryIdentityExpired) {
+                router && router.push("/account/logins/email/new");
```

Comment:
Yes, I didn't really knew what to do exactly in this use case. But since the validation code (& thus `TemporaryIdentity` on our side) is expired, user would need to enter a new one. A flash message would be nice too, but I think it would be easier to implement when we will have flash message handled on server side, no ?

Reply:
Hmm, yes you're right